### PR TITLE
API: Datetime/TimedeltaArray from to_datetime

### PR DESCRIPTION
**Pull request by TomAugspurger**
Closes https://github.com/pandas-dev/pandas/issues/24656

### CODE REVIEW

**Comment by TomAugspurger** [comment]
cc @jorisvandenbossche.

Do the docs make sense, or do they just cause more confusion?

File: pandas/core/arrays/array_.py [hunk 1]
```diff
@@ -239,5 +251,14 @@ def array(data,         # type: Sequence[object]
 
         # TODO(BooleanArray): handle this type
 
+    # Pandas overrides NumPy for
+    #   1. datetime64[ns]
+    #   2. timedelta64[ns]
+    # so that a DatetimeArray is returned.
+    if is_datetime64_ns_dtype(dtype):
+        return DatetimeArray._from_sequence(data, dtype=dtype, copy=copy)
+    if is_timedelta64_ns_dtype(dtype):
```
Comment (jreback) [inline pandas/core/arrays/array_.py]: elif

File: pandas/core/arrays/array_.py [hunk 2]
```diff
@@ -120,6 +122,16 @@ def array(data,         # type: Sequence[object]
     ['a', 'b']
     Length: 2, dtype: str32
 
+    Finally, Pandas has arrays that mostly overlap with NumPy
```
Comment (jreback) [inline pandas/core/arrays/array_.py]: can you add an example with datetime64[ns]

**Review by jreback — CHANGES_REQUESTED** [review]
small comments otherwise lgtm.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm merge on green.

File: pandas/tests/arrays/test_array.py [hunk 3]
```diff
@@ -36,8 +36,30 @@
...
-     PandasArray(np.array([1, 2], dtype='datetime64[ns]'))),
-    # TODO(DatetimeArray): add here
+     pd.arrays.DatetimeArray._from_sequence(
+         np.array([1, 2], dtype='datetime64[ns]'))),
+
+    (np.array([1, 2], dtype='datetime64[ns]'), None,
+     pd.arrays.DatetimeArray._from_sequence(
+         np.array([1, 2], dtype='datetime64[ns]'))),
+
+    (pd.DatetimeIndex(['2000', '2001']), np.dtype('datetime64[ns]'),
```
Comment (jorisvandenbossche) [inline pandas/tests/arrays/test_array.py]: this one also with a dtype of None ?

Comment (TomAugspurger) [reply pandas/tests/arrays/test_array.py]: 351c781c55b5391833ab8cccf23669226cc8de52

**Pull request merged**